sender.go

```diff
@@ -121,116 +87,118 @@ func (s *Sender) SendNoRetry(msg *Message) (*Response, error) {
 //
 // Note that messages are retried using exponential backoff, and as a
 // result, this method may block for several seconds.
-func (s *Sender) Send(msg *Message, retries int) (*Response, error) {
-	if err := checkSender(s); err != nil {
-		return nil, err
-	} else if err := checkMessage(msg); err != nil {
-		return nil, err
+func (s *Sender) Send(msg *messaging.MulticastMessage, retries int) (*messaging.BatchResponse, []string, error) {
+	// Note that failed tokens returns as nil, since we cannot guarantee that msg.Tokens exists
+	if err := checkMessage(msg); err != nil {
+		return nil, nil, err
+	} else if err = checkSender(s); err != nil {
+		return nil, msg.Tokens, err
 	} else if retries < 0 {
```

Comment:
Could we just change these if...else if's into if statements? Just a nitpick observation. The same applies for `SendNoRetry()`

Reply:
On second thought: this PR aims to keep everything as similar as before to avoid breaking changes, so lets leave it is as.